src/halo-hooks/useRewards.ts

```diff
@@ -307,3 +312,58 @@ export const useRewarderUSDPrice = (rewarderAddress: string | undefined) => {
 
   return rewarderTokenUsdPrice[rewarderTokenAddress[0]]
 }
+
+export const useGetBaseApr = (poolAddress: string, tokenPair: string): number => {
+  const { account, library, chainId } = useActiveWeb3React()
+  const [baseAPR, setBaseApr] = useState(0)
+  const poolsWithBaseAPR = Object.keys(DEPOSIT_TXHASH)
+
+  const fetchBaseUsdHlpPrice = useCallback(async () => {
+    let initialUsdHlpPrice = 0
+    let currentUsdHlpPrice = 0
+    let quoteTokenInitialUsdPrice = 0
+    let baseTokenInitialUsdPrice = 0
+    let hlpMintedValue = 0
+    if (!library || !poolsWithBaseAPR.includes(poolAddress.toLowerCase())) return
+    const txReceipt = await library.getTransactionReceipt(DEPOSIT_TXHASH[poolAddress.toLowerCase()])
+    const curveContract = getContract(poolAddress, CURVE_ABI, library, account ?? undefined)
+    const date = getTxDate((await library.getBlock(txReceipt.blockNumber)).timestamp)
+
+    // get the quote token info
+    const baseToken = tokenPair.split('/')[0]
+    const baseTokenInfo = (haloTokenList[chainId as ChainId] as Token[]).filter(token => token.symbol === baseToken)[0]
+
+    // Calculate initial USD:HLP price during the first deposit tx of the pool
+    baseTokenInitialUsdPrice =
+      parseFloat(ethers.utils.formatUnits(txReceipt.logs[0].data, baseTokenInfo.decimals)) *
+      (await getInitialTokenUSDPrice(TOKEN_COINGECKO_NAME[baseToken.toLowerCase()], date))
+    if (baseTokenInfo.symbol === 'XSGD') {
+      quoteTokenInitialUsdPrice =
+        parseFloat(ethers.utils.formatUnits(txReceipt.logs[1].data, 6)) *
+        (await getInitialTokenUSDPrice(TOKEN_COINGECKO_NAME['usdc'], date))
+      hlpMintedValue = parseFloat(ethers.utils.formatUnits(txReceipt.logs[2].data, 18))
+    } else {
+      quoteTokenInitialUsdPrice =
+        parseFloat(ethers.utils.formatUnits(txReceipt.logs[2].data, 6)) *
+        (await getInitialTokenUSDPrice(TOKEN_COINGECKO_NAME['usdc'], date))
+      hlpMintedValue = parseFloat(ethers.utils.formatUnits(txReceipt.logs[3].data, 18))
+    }
+    initialUsdHlpPrice = (baseTokenInitialUsdPrice + quoteTokenInitialUsdPrice) / hlpMintedValue
+
+    // Calculate current USD:HLP price
+    const [curveLiquidity, curveTotalSupply] = await Promise.all([
+      curveContract.liquidity(),
+      curveContract.totalSupply()
+    ])
+    currentUsdHlpPrice = parseFloat(formatEther(curveLiquidity.total_)) / parseFloat(formatEther(curveTotalSupply))
+
+    // Calculate the Base APR
+    setBaseApr((initialUsdHlpPrice / currentUsdHlpPrice - 1) * (365 / 190) * 100)
```

Comment:
@geonellov we shouldn't hardcode 190 here. 

this should be `Base APR = ( ( initial_USD_HLP_price / current_USD_HLP_price ) - 1 ) * (365 days / # of days since first deposit tx) * 100 = % value`

so instead of `190` it should be `<# of days since first deposit tx>`